django/db/models/sql/query.py

```diff
@@ -1033,6 +1035,10 @@ def resolve_expression(self, query, *args, **kwargs):
                 not self.select_for_update):
             clone.clear_ordering(True)
         clone.where.resolve_expression(query, *args, **kwargs)
+        # Resolve combined queries.
+        for combined_query in clone.combined_queries:
+            combined_query.bump_prefix(query)
+            combined_query.where.resolve_expression(query, *args, **kwargs)
```

Comment:
I'm curious as to whether calling `combined_query.resolve_expression(query, *args, **kwargs)` instead to trigger a recursive resolving would work here?

```python
clone.combined_queries = tuple(
    query.resolve_expression(query, *args, **kwargs) for query in clone.combined_queries
)
```

Reply:
I can confirm that this works :+1: